yt_dlp/extractor/niconico.py

```diff
@@ -470,87 +467,26 @@ def get_video_info(*items, get_first=True, **kwargs):
                 parse_duration(self._html_search_meta('video:duration', webpage, 'video duration', default=None))
                 or get_video_info('duration')),
             'webpage_url': url_or_none(url) or f'https://www.nicovideo.jp/watch/{video_id}',
-            'subtitles': self.extract_subtitles(video_id, api_data, session_api_data),
+            'subtitles': self.extract_subtitles(video_id, api_data),
         }
 
-    def _get_subtitles(self, video_id, api_data, session_api_data):
-        comment_user_key = traverse_obj(api_data, ('comment', 'keys', 'userKey'))
-        user_id_str = session_api_data.get('serviceUserId')
-
-        thread_ids = traverse_obj(api_data, ('comment', 'threads', lambda _, v: v['isActive']))
-        legacy_danmaku = self._extract_legacy_comments(video_id, thread_ids, user_id_str, comment_user_key) or []
-
+    def _get_subtitles(self, video_id, api_data):
         new_comments = traverse_obj(api_data, ('comment', 'nvComment'))
         new_danmaku = self._extract_new_comments(
             new_comments.get('server'), video_id,
             new_comments.get('params'), new_comments.get('threadKey'))
```

Comment:
We could inline the `_extract_new_comments` code in this method. Seems like an unnecessary abstraction now. Could also remove references to "new"

Reply:
Fixed in 5e1fffc5adc02de43d962eeae48ba0a9b9348e98 .